src/NServiceBus.Core/Transports/Msmq/Config/MsmqTransportConfigurator.cs

```diff
@@ -163,10 +112,50 @@ protected override void Configure(FeatureConfigurationContext context, string co
             }
 
             var messageLabelGenerator = context.Settings.GetMessageLabelGenerator();
-            context.Container.ConfigureComponent(b=>new MsmqMessageSender(settings, messageLabelGenerator), DependencyLifecycle.InstancePerCall);
+            context.Container.ConfigureComponent(b => new MsmqMessageSender(settings, messageLabelGenerator), DependencyLifecycle.InstancePerCall);
 
             context.Container.ConfigureComponent<MsmqQueueCreator>(DependencyLifecycle.InstancePerCall)
                 .ConfigureProperty(t => t.Settings, settings);
+
+
+            if (context.Settings.GetOrDefault<bool>("Endpoint.SendOnly"))
+            {
+                return;
+            }
+
+            //todo: pass this in the deque settings instead
+            var transactionSettings = new Unicast.Transport.TransactionSettings(context.Settings);
+
+            context.Container.ConfigureComponent(b =>
+            {
+                MsmqReceiveStrategy receiveStrategy;
```

Comment:
This lambda inside the builder makes me nervous ;) Should we add the factory method or a factory again?